packages/aws-cdk-lib/custom-resources/lib/provider-framework/waiter-state-machine.ts

```diff
@@ -49,6 +92,32 @@ export class WaiterStateMachine extends Construct {
     props.isCompleteHandler.grantInvoke(role);
     props.timeoutHandler.grantInvoke(role);
 
+    let logGroup: LogGroup | undefined;
+    if (props.disableLogging) {
+      if (props.logOptions) {
+        throw new Error('logOptions must not be used if disableLogging is true');
+      }
+    } else {
+      logGroup = new LogGroup(this, 'LogGroup', {
+        retention: props.logOptions?.logRetention,
+      });
+      role.addToPrincipalPolicy(new PolicyStatement({
+        actions: [
+          'logs:CreateLogDelivery',
+          'logs:CreateLogStream',
+          'logs:GetLogDelivery',
+          'logs:UpdateLogDelivery',
+          'logs:DeleteLogDelivery',
+          'logs:ListLogDeliveries',
+          'logs:PutLogEvents',
+          'logs:PutResourcePolicy',
+          'logs:DescribeResourcePolicies',
+          'logs:DescribeLogGroups',
+        ],
+        resources: ['*'],
+      }));
+    }
```

Comment:
`The following is an example policy you can use to configure your permissions. As shown in the following example, you need to specify * in the Resource field because CloudWatch API actions, such as CreateLogDelivery and DescribeLogGroups, don't support [Resource types defined by Amazon CloudWatch Logs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncloudwatchlogs.html#amazoncloudwatchlogs-resources-for-iam-policies).`

https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html#cloudwatch-iam-policy